### ISSUE TITLE
Implicit Terraform Cloud workspace creation fails

### ISSUE THREAD

**Issue by ritsok** [hashicorp/terraform#28092]
### Terraform Version

```
0.14.8
```

### Terraform Configuration Files
```terraform
 terraform {
  # backend "remote" {
  #   organization = "hashicorp-learn"
  #   workspaces {
  #     name = "example_workspace"
  #   }
  # }

  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 3.27"
    }
  }
}

provider "aws" {
  profile = "default"
  region  = "us-west-2"
}

resource "aws_instance" "example" {
  ami           = "ami-08d70e59c07c61a3a"
  instance_type = "t2.micro"
}

```

### Expected Behavior

Per our [docs](https://www.terraform.io/docs/cloud/run/cli.html#implicit-workspace-creation), Terraform should create the example workspace.

### Actual Behavior

`terraform init` errors out:
```
➜  learn-terraform-aws-instance terraform init

Initializing the backend...

Error: Error looking up workspace

Workspace read failed: resource not found
```

### Steps to Reproduce

1. Run `terraform apply` with the above configuration, which will use a local backend for the resources
2. Uncomment the remote backend configuration
3. Run `terraform init`

### Additional Context
- As a troubleshooting attempt, I have tried to remove the `.terraform` cache directory. It doesn't fix it
- Toggling between terraform versions *does* fix it. Still getting an error with `0.14.{7,6,5}`, but it works with `0.14.4`

**Comment by alisdair** [comment]
Hi @ritsok, thanks for reporting this! I can confirm this behaviour on 0.14.8, and also using a simpler configuration with your steps:

```hcl
 terraform {
  # backend "remote" {
  #   organization = "alisdair-v2"
  #   workspaces {
  #     name = "28092"
  #   }
  # }

  required_providers {
    random = {
      source  = "hashicorp/random"
      version = "~> 3.0.0"
    }
  }
}

resource "random_pet" "pet" {
  length = 3
}
```

1. `terraform init`
2. `terraform apply -auto-approve`
3. Uncomment the remote backend configuration
4. **Make sure your Terraform Cloud organization does not have a workspace with this name**
5. `terraform init` fails with "Workspace read failed: resource not found"

Good sleuthing with the Terraform versions! As a result I'm fairly sure that the underlying cause of this bug is #27545, which added a version check when migrating state to a Terraform Cloud workspace. We'll need to fix that to cope with a missing workspace.

**Comment by alisdair** [comment]
Thanks again for the report! The fix is in the upcoming 0.14.9 patch release, as well as the 0.15 beta series.
